### DISCUSSION TITLE
how to set timeout when exec client connect?

### DISCUSSION THREAD

**Discussion by caolf** [gijzelaerr/python-snap7#556] [Q&A] [answered]
It spends about one minute when exec 
`snap7.client.Client().connect(self._config.address, self._config.rack, self._config.slot, self._config.port)`  to revcived timeout .
how to change timeout time and set timeout 10s ?

Look forward to your help, thank you

**Comment by Alberttorm9** [comment]
Maybe I'm late but for people who still need it:

from snap7.type import Parameter

PlcClient= snap7.client.Client()
plc.PlcClient(Address, rack, slot)

PlcClient.set_param(Parameter.RecvTimeout, 10000) <-- In ms

Can use:
print(PlcClient.set_param(Parameter.RecvTimeout))

to know if it changed

**Comment by gijzelaerr** [comment, marked as answer]
Closing as answered. A solution using `set_param(Parameter.RecvTimeout, value_in_ms)` was provided. In python-snap7 3.0, timeout can also be configured directly.
